card/mod/01_core/set/all/collection.rb

```diff
@@ -175,38 +175,55 @@ def set_search_params_variables! hash
       end
     end
   end
+
+
+  def each_nest args={}
+    Card::Content.new(card.content, card).find_chunks( Card::Chunk::Reference ).each do |chunk|
+      yield(chunk.referee_name.to_s, nest_args(args,chunk))
+    end
+  end
+
+  def map_nests args={}, &block
+    result = []
+    each_nest args do |name, n_args|
+      result << block.call(name, n_args)
+    end
+    result
+  end
+
+  # process args for links and nests
+  def nest_args args, chunk=nil
+    r_args = item_args(args)
+    if @inclusion_opts
+      r_args.merge! @inclusion_opts.clone
+    end
+    if chunk.kind_of? Card::Chunk::Include
+      r_args.merge!(chunk.options)
+    elsif chunk.kind_of? Card::Chunk::Link
+      r_args.reverse_merge!(:view=>:link)
+      r_args.reverse_merge!(:title=>chunk.link_text) if chunk.link_text
+    end
```

Comment:
good looking code.  at some point in the future perhaps we may be able to move some of this logic into Chunk classes?